esphome/components/vl53l1x/sensor.py

```diff
@@ -0,0 +1,236 @@
+from esphome import pins
+import esphome.codegen as cg
+from esphome.components import i2c, sensor
+import esphome.config_validation as cv
+from esphome.const import (
+    CONF_ADDRESS,
+    CONF_ENABLE_PIN,
+    CONF_INTERRUPT_PIN,
+    CONF_OFFSET,
+    CONF_UPDATE_INTERVAL,
+    CONF_X,
+    CONF_Y,
+    DEVICE_CLASS_DISTANCE,
+    ICON_ARROW_EXPAND_VERTICAL,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_METER,
+)
+
+DEPENDENCIES = ["i2c"]
+
+vl53l1x_ns = cg.esphome_ns.namespace("vl53l1x")
+VL53L1xSensor = vl53l1x_ns.class_(
+    "VL53L1xSensor", sensor.Sensor, cg.Component, i2c.I2CDevice
+)
+
+CONF_TIMING_BUDGET = "timing_budget"
+CONF_DISTANCE_MODE = "distance_mode"
+CONF_XTALK_CORRECTION = "xtalk_correction"
+CONF_DISTANCE_THRESHOLD = "distance_threshold"
+CONF_MIN = "min"
+CONF_MAX = "max"
+CONF_INTERRUPT_WHEN = "interrupt_when"
+CONF_REGION_OF_INTEREST = "region_of_interest"
+CONF_W = "w"
+CONF_H = "h"
+CONF_SIGNAL_THRESHOLD = "signal_threshold"
+CONF_SIGMA_THRESHOLD = "sigma_threshold"
+
+DISTANCE_MODE_ENUM = vl53l1x_ns.enum("DistanceMode")
```

Comment:
```suggestion
DistanceMode = vl53l1x_ns.enum("DistanceMode")
```
PascalCase for C `MockObj` variables seems to be the preferred style across the codebase.
Same for the other enum below.